File: src/main/java/com/drodriguln/domain/ArtistRepository.java
package com.drodriguln.domain;

import com.drodriguln.model.Artist;
import org.springframework.data.mongodb.repository.MongoRepository;
import org.springframework.stereotype.Repository;

@Repository
public interface ArtistRepository extends MongoRepository<Artist, String> {
    Artist findByName(String name);
}

File: src/main/kotlin/com/drodriguez/maestro/api/domain/ArtistRepository.kt
package com.drodriguez.maestro.api.domain

import com.drodriguez.maestro.api.model.Artist
import org.springframework.data.mongodb.repository.MongoRepository
import org.springframework.stereotype.Repository

@Repository
interface ArtistRepository : MongoRepository<Artist, String> {
    fun findByName(name: String): Artist
}